### PR TITLE
updated base component card and grid

### PR DESCRIPTION
**Pull request by odedre**
## Proposed Changes

- grid component now accepts className from outside
- base component card separated to independent modules
- base component card updated to use modules
- moved deprecated sticker to right corner of preview

### CODE REVIEW

File: scopes/explorer/ui/base-component-card/base-component-card.tsx [hunk 1]
```diff
@@ -39,41 +50,18 @@ export function BaseComponentCard({
...
-            </div>
-            <div className={styles.description}>{description}</div>
-          </div>
-          {children}
-        </div>
+    <Card className={className}>
+      <Link href={id} className={styles.link}>
+        <DeprecationSticker isDeprecated={isDeprecated} />
+        <PreviewContainer preview={preview} />
+        <div className={styles.avatarContainer}>{avatar}</div>
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/base-component-card.tsx:61]: check if avatar can be removed

File: scopes/explorer/ui/base-component-card/base-component-card.tsx [hunk 2]
```diff
@@ -29,6 +36,10 @@ export type BaseComponentCardProps = {
    * true if the component is deprecated
    */
   isDeprecated?: boolean;
+  /**
+   * avatar component
+   */
+  avatar?: ComponentType;
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/base-component-card.tsx:42]: type should be ReactNode

File: scopes/explorer/ui/base-component-card/base-component-card.tsx [hunk 3]
```diff
@@ -29,6 +36,10 @@ export type BaseComponentCardProps = {
...
 } & React.HTMLAttributes<HTMLDivElement>;
 
 export function BaseComponentCard({
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/base-component-card.tsx:45]: please consider remove this and use building-blocks directly

File: scopes/explorer/ui/base-component-card/card/card.module.scss [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.componentCard {
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/card/card.module.scss:1]: consider using base-ui.surfaces.card

File: scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.deprecated {
+  display: none;
+  position: absolute;
+  top: 7px;
+  right: 12px;
+  background-color: #ffc640;
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss:6]: --base-color + colorPalette.hunger

File: scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
+.deprecated {
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss:1]: consider using a container that will be the take care of the positioning


```scss
	.stickersBox {
		position: absolute;
		
		display: flex;
		justify-content: start;
		align-items: center;
		margin: $size6px 0 0 $size6px;
		
		> *:not(:last-child) {
			margin-right: $size5px;
		}
		
		&:empty {
			display: none;
		}
	}
```

File: scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+export type DeprecationStickerProps = {
+  isDeprecated?: boolean;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+export function DeprecationSticker({ isDeprecated }: DeprecationStickerProps) {
+  return (
+    <div
+      className={classNames(styles.deprecated, {
+        [styles.show]: isDeprecated,
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.tsx:14]: isDeprecated && styles.show

File: scopes/explorer/ui/base-component-card/details/details.module.scss [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    white-space: nowrap;
+    margin-right: 5px;
+  }
+}
+.description {
+  color: var(--text-muted, #6c707c);
+  font-size: var(--p-xs, 14px);
+  overflow: hidden;
+  text-overflow: ellipsis;
+  display: -webkit-box;
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/details/details.module.scss:36]: check in other browser

Comment (KutnerUri) [reply scopes/explorer/ui/base-component-card/details/details.module.scss:36]: use the `line-clamp` property, or add all the other prefixes

File: scopes/explorer/ui/base-component-card/details/details.module.scss [hunk 9]
```diff
@@ -0,0 +1,39 @@
+.content {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  padding: 16px;
+  height: 96px;
+}
+.nameSpace {
+  color: var(--text-muted, #6c707c);
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/details/details.module.scss:9]: consider using the mutedText class

File: scopes/explorer/ui/base-component-card/details/details.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import styles from './details.module.scss';
+
+export type ComponentDetailsProps = {
+  id: string;
+  version?: string;
+  description?: string;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+export function ComponentDetails({ id, version, description, className, ...rest }: ComponentDetailsProps) {
+  const idArray = id.split('/');
```
Comment (KutnerUri) [inline scopes/explorer/ui/base-component-card/details/details.tsx:13]: at least add comment to use component id

**Review by KutnerUri — APPROVED** [review]

File: scopes/explorer/ui/base-component-card/base-component-card.tsx (hunk 1 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/base-component-card.tsx:61]: removed

File: scopes/explorer/ui/base-component-card/base-component-card.tsx (hunk 3 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/base-component-card.tsx:45]: done

File: scopes/explorer/ui/base-component-card/base-component-card.tsx (hunk 2 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/base-component-card.tsx:42]: removed avatar

File: scopes/explorer/ui/base-component-card/card/card.module.scss (hunk 4 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/card/card.module.scss:1]: card has built in padding that I will need to override. lets talk about this later

File: scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss (hunk 5 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/deprecation-sticker/deprecation-sticker.module.scss:6]: this is overly complex and hard to use. lets talk about it

File: scopes/explorer/ui/base-component-card/details/details.tsx (hunk 10 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/details/details.tsx:13]: done

File: scopes/explorer/ui/base-component-card/details/details.module.scss (hunk 8 above)
Comment (odedre) [reply scopes/explorer/ui/base-component-card/details/details.module.scss:36]: weirdly it works on safari and firefox as well without any extra prefixes

**Pull request merged**